src/Tools/Pagination/Paginator.php

```diff
@@ -154,6 +154,35 @@ public function getIterator(): Traversable
         return new ArrayIterator($result);
     }
 
+    /**
+     * Returns Query prepared to count.
+     */
+    protected function getCountQuery(): Query
```

Comment:
```suggestion
    final protected function getCountQuery(): Query
```

I don't think you actually need to override that method, right? Being able to call it from your `count()` override and then modifying the result sounds like it would be sufficient.